ResearchStack2/ResearchStack2/RSDUIActionType.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  RSDUIActionType.swift
+//  ResearchStack2
+//
+//  Copyright © 2018 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// The `RSDUIActionType` enum describes standard navigation actions that are common to a
+/// given UI step. It is extendable using the custom field.
+public enum RSDUIActionType {
+    
+    /// Standard navigation elements that are common to most steps.
+    case navigation(Navigation)
+    
+    /// Standard navigation elements that are common to most steps.
```

Comment:
is this comment being identical to the one above intentional?

Reply:
Yes. The way this enum is set up there are effectively two "categories" of actions.

1. Navigation - case where there are standard elements that are supported by the base class implementation of RSDStepViewController.
2. Custom - case where an app defines custom keywords that the app is responsible for managing.

It is "subtly" different from an extensible string enums that are used elsewhere because it contains a subset that are explicitly defined and expected to be handled by the UI.